source/kernel/Makefile

```
@@ -172,17 +173,28 @@ everything:
+# Check that the required tools are available. This is done at parse time
+# (not in a recipe) so that it runs before anything else regardless of the
+# make version: the former "-include prerequisites" trick relied on make
+# running a phony target's recipe at startup, which GNU make 4.4+ no longer
+# does. Skipped when just cleaning.
+ifneq ($(MAKECMDGOALS),clean)
+$(foreach exec,$(TOOLS),\
         $(if $(shell which $(exec)),,$(error "ERROR: can't execute $(exec), is it installed/in PATH?")))
+endif
 
--include prerequisites
+base: $(PUBLISHED_BASE)
```

checkmake (0.3.2)

[warning] Target "base" should be declared PHONY.

(phonydeclared)